instrumentation/rxjava-2.0/library/src/main/java/io/opentelemetry/instrumentation/rxjava2/TracingAssembly.java

```diff
@@ -0,0 +1,266 @@
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.opentelemetry.instrumentation.rxjava2;
+
+import io.opentelemetry.context.Context;
+import io.opentelemetry.context.Scope;
+import io.reactivex.Completable;
+import io.reactivex.CompletableObserver;
+import io.reactivex.Flowable;
+import io.reactivex.Maybe;
+import io.reactivex.MaybeObserver;
+import io.reactivex.Observable;
+import io.reactivex.Observer;
+import io.reactivex.Single;
+import io.reactivex.SingleObserver;
+import io.reactivex.functions.BiFunction;
+import io.reactivex.functions.Function;
+import io.reactivex.internal.fuseable.ConditionalSubscriber;
+import io.reactivex.parallel.ParallelFlowable;
+import io.reactivex.plugins.RxJavaPlugins;
+import org.checkerframework.checker.lock.qual.GuardedBy;
+import org.reactivestreams.Subscriber;
+
+/**
+ * RxJava2 library instrumentation.
+ *
+ * <p>In order to enable RxJava2 instrumentation one has to call the <code>TracingAssembly.enable()
+ * </code> method.
+ *
+ * <p>Instrumentation uses <code>on*Assembly</code> and <code>on*Subscribe</code> RxJavaPlugin hooks
+ * to wrap RxJava2 classes in their tracing equivalents.
+ *
+ * <p>Instrumentation can be disabled by calling the <code>TracingAssembly.disable()</code> method.
+ */
+public final class TracingAssembly {
+
+  @SuppressWarnings("rawtypes")
+  @GuardedBy("TracingAssembly.class")
+  private static BiFunction<? super Observable, ? super Observer, ? extends Observer>
+      oldOnObservableSubscribe;
+
+  @SuppressWarnings("rawtypes")
+  @GuardedBy("TracingAssembly.class")
+  private static BiFunction<
+          ? super Completable, ? super CompletableObserver, ? extends CompletableObserver>
+      oldOnCompletableSubscribe;
+
+  @SuppressWarnings("rawtypes")
+  @GuardedBy("TracingAssembly.class")
+  private static BiFunction<? super Single, ? super SingleObserver, ? extends SingleObserver>
+      oldOnSingleSubscribe;
+
+  @SuppressWarnings("rawtypes")
+  @GuardedBy("TracingAssembly.class")
+  private static BiFunction<? super Maybe, ? super MaybeObserver, ? extends MaybeObserver>
+      oldOnMaybeSubscribe;
+
+  @SuppressWarnings("rawtypes")
+  @GuardedBy("TracingAssembly.class")
+  private static BiFunction<? super Flowable, ? super Subscriber, ? extends Subscriber>
+      oldOnFlowableSubscribe;
+
+  @SuppressWarnings("rawtypes")
+  @GuardedBy("TracingAssembly.class")
+  private static Function<? super ParallelFlowable, ? extends ParallelFlowable>
+      oldOnParallelAssembly;
+
+  @GuardedBy("TracingAssembly.class")
+  private static boolean enabled;
+
+  private TracingAssembly() {}
+
+  public static synchronized void enable() {
+    if (enabled) {
+      return;
+    }
+
+    enableObservable();
+
+    enableCompletable();
+
+    enableSingle();
+
+    enableMaybe();
+
+    enableFlowable();
+
+    enableParallel();
```

Comment:
Minor nitpicking: consider keeping the same order in fields, enable & disable methods

Reply:
👍